modules/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureBlobContainerRetriesTests.java

```diff
@@ -72,6 +75,48 @@ public void testReadNonexistentBlobThrowsNoSuchFileException() {
         assertThat(exception.toString(), exception.getMessage().toLowerCase(Locale.ROOT), containsString("not found"));
     }
 
+    public void testReadBlobBeyondRangeThrowsRequestedRangeNotSatisfiedException() throws IOException {
```

Comment:
Hmm, doesn't seem like this test fits in the "Retries" tests for Azure.

Should we better add support for returning `RestStatus.REQUESTED_RANGE_NOT_SATISFIED` in `AzureHttpHandler.java`, similar to how it's done in `S3HttpHandler`, and then add a test like `testReadFromPositionLargerThanBlobLength` (which simply does an invalid request) to `AzureStorageCleanupThirdPartyTests`?

Reply:
Agree. This adds to testing code clarity